StRoot/StEvent/StGmtStripCollection.h

```diff
@@ -0,0 +1,155 @@
+    size_t getNumStrips() const;
+    
+    // modify/access the moduleId
+    short getModule() const;
+    void setModule( short module );
+    
+    // Clear
+    void Clear( Option_t *opt = "" );
+    
+    // Get pointer to a strip -- note: this is the only way to modify a
+    // strip.  New strip is created if it does not exist, but only
+    // using StGmtStrip() constructor.  Ownership is retained by the
+    // collection.
+//     StGmtStrip* getStrip( int elecId );
+    StGmtStrip* getStrip( int Id );
+    StGmtStrip* getSortedStrip( int Id );
+    
+ protected:  
+  /// Function used for sorting strips by geoId
+  static bool hitGeoIdLessThan( const StGmtStrip* h1, const StGmtStrip* h2 ); 
+  /// Function used for sorting strips by coordinate number
+  static bool hitCoordLessThan( const StGmtStrip* h1, const StGmtStrip* h2 ); 
+  /// Function used for sorting strips by X then Y
+  static bool hitLayerLessThan( const StGmtStrip* h1, const StGmtStrip* h2 ); 
+     
+  /// Module ID
+  Short_t mModule;
+  /// Vector with strips
+  StSPtrVecGmtStrip mStripVec;    
+    
+  /// Temporary copy of the pointers, indexed by ElecId 
+  /// used for the addStripInfo class
+  StPtrVecGmtStrip mStripElecIdVec;
+  /// Temporary copy of the pointers, indexed by GeoId 
+  /// used for the addStripInfo class
+  StPtrVecGmtStrip mStripGeoIdVec; 
+    
+ private:   
+  ClassDef(StGmtStripCollection,1)
+}; 
+
+
+// inline functions
+
+// inline StGmtStripCollection::StGmtStripCollection( short module ) : StObject(), mModule( module ) {
+//     mStripElecIdVec.resize( kGmtNumElecIds );
+//     for (unsigned int i=0; i<mStripElecIdVec.size(); i++)
+//         mStripElecIdVec[i] = static_cast< StGmtStrip* >(0);
+// };
+
+inline StGmtStripCollection::StGmtStripCollection( short module ) : StObject(), mModule( module ) {
+    mStripGeoIdVec.resize( kGmtNumGeoIds );
+    for (unsigned int i=0; i<mStripGeoIdVec.size(); i++)
+        mStripGeoIdVec[i] = static_cast< StGmtStrip* >(0);
+};
+
+// inline StGmtStripCollection::StGmtStripCollection( short module ) : StObject(), mModule( module ) {
+//     mStripVec.resize( kGmtNumGeoIds );
+//     for (unsigned int i=0; i<mStripVec.size(); i++)
+//         mStripVec[i] = static_cast< StGmtStrip* >(0);
+// };
+
+
+// sort by geoId
+inline void StGmtStripCollection::sortByGeoId(){
+    sort( mStripVec.begin(), mStripVec.end(), &StGmtStripCollection::hitGeoIdLessThan );
+    return;
+};
+
+// sort by layer (X first then Y)
+inline void StGmtStripCollection::sortByLayer(){
+    sort( mStripVec.begin(), mStripVec.end(), &StGmtStripCollection::hitLayerLessThan );
+    return;
+};
+
+// sort by coordinate number
+inline void StGmtStripCollection::partialSortByCoord(){
+    partial_sort( mStripVec.begin(), mStripVec.begin()+kGmtNumStrips, mStripVec.begin()+kGmtNumStrips, &StGmtStripCollection::hitCoordLessThan );
+    return;
+};
+
+// sort by coordinate number
+inline void StGmtStripCollection::sortByCoord(){
```

Comment:
Just a comment.  You can always sort vectors inside of consumer codes.  So... I don't really see a need to sort the in-memory collection, unless there are multiple downstream codes that would prefer a given order.